### PR TITLE
Simplify and fix example for the StackPanel vs DockPanel article

### LINKED ISSUE #1586: A picture or video would be nice considering this example code wont work without a bunch of modifications...
... so we can at least see if the thing we hack together based on this code looks like you expect it to look.

Please make the XAML (Page) to be consistent with the c# (Window)

Its also not clear that the two blocks of code should be used as one or the other, not both.  Where does the c# code go in a blank WPF app?

The c# code as it is produces 25 compiler errors.  And thats not counting the runtime errors because the smiley png does not exist anywhere.
image

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 2d210673-78b4-3eb7-d949-3fdc492c0caa
* Version Independent ID: 55957ba3-8779-5965-804c-17849e3513df
* Content: [How to: Choose Between StackPanel and DockPanel - WPF .NET Framework](https://learn.microsoft.com/en-us/dotnet/desktop/wpf/controls/how-to-choose-between-stackpanel-and-dockpanel?view=netframeworkdesktop-4.8)
* Content Source: [dotnet-desktop-guide/framework/wpf/controls/how-to-choose-between-stackpanel-and-dockpanel.md](https://github.com/dotnet/docs-desktop/blob/main/dotnet-desktop-guide/framework/wpf/controls/how-to-choose-between-stackpanel-and-dockpanel.md)
* Product: **dotnet-framework**
* Technology: **dotnet-wpf**
* GitHub Login: @adegeo
* Microsoft Alias: **adegeo**

---
[Associated WorkItem - 69674](https://dev.azure.com/msft-skilling/Content/_workitems/edit/69674)

### PR DESCRIPTION
**Pull request by adegeo**
## Summary

- Replaced the image resource required by anyone using this code with an emoji
- Simplified the code-first approach
- Split example into sections to clarify the difference between the XAML version and code-first version
- Added an image to the doc

@StingyJack if you're watching, here is the updated [csharp code](https://github.com/dotnet/docs-desktop/pull/1600/files#diff-a232a5d7be7060d89a2be03ece5342c7ba6622f9b5ce0d1d6572fa1ee4cc926cR29-R78), [vb.net code](https://github.com/dotnet/docs-desktop/pull/1600/files#diff-dc7e83424c12232da8c39a564bb51e5ade345eee98ad061aea9c3fb6a726479fR3-R45), and [xaml example](https://github.com/dotnet/docs-desktop/pull/1600/files#diff-5ed469577e76a1976b71795b9588e6e1a5037c579af28bc6e6b71373f6803ba2R10-R36).

Fixes #1586

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [dotnet-desktop-guide/framework/wpf/controls/how-to-choose-between-stackpanel-and-dockpanel.md](https://github.com/dotnet/docs-desktop/blob/2042678c3d721b648af68796ca00847e2e114be3/dotnet-desktop-guide/framework/wpf/controls/how-to-choose-between-stackpanel-and-dockpanel.md) | [dotnet-desktop-guide/framework/wpf/controls/how-to-choose-between-stackpanel-and-dockpanel](https://review.learn.microsoft.com/en-us/dotnet/desktop/framework/wpf/controls/how-to-choose-between-stackpanel-and-dockpanel?branch=pr-en-us-1600) |

### CODE REVIEW

**Comment by adegeo** [comment]
@luisquintanilla Ready for review

File: dotnet-desktop-guide/framework/wpf/controls/how-to-choose-between-stackpanel-and-dockpanel.md [hunk 1]
```diff
@@ -15,18 +14,24 @@ ms.assetid: f9239086-451f-42e6-81f7-ef89ef349742
...
- [!code-csharp[StackPanelOvw4#1](~/samples/snippets/csharp/VS_Snippets_Wpf/StackPanelOvw4/CSharp/StackPanel_Ovw_Sample4.cs#1)]
- [!code-vb[StackPanelOvw4#1](~/samples/snippets/visualbasic/VS_Snippets_Wpf/StackPanelOvw4/VisualBasic/StackPanelSamp.vb#1)]
- [!code-xaml[StackPanelOvw4#1](~/samples/snippets/xaml/VS_Snippets_Wpf/StackPanelOvw4/XAML/default.xaml#1)]
+The following example demonstrates this key difference between <xref:System.Windows.Controls.DockPanel> and <xref:System.Windows.Controls.StackPanel> when designing a page in XAML.
+
+:::code language="xaml" source="./snippets/how-to-choose-between-stackpanel-and-dockpanel/xaml/MainWindow.xaml" id="EmojiViewBox":::
+
+## Code-based example
+
+The following example demonstrates this key difference between <xref:System.Windows.Controls.DockPanel> and <xref:System.Windows.Controls.StackPanel> when setting the content of a `Window` during the `Window.Loaded` event:
```
Comment (adegeo) [inline dotnet-desktop-guide/framework/wpf/controls/how-to-choose-between-stackpanel-and-dockpanel.md:31]:
```suggestion
The following example demonstrates this key difference between <xref:System.Windows.Controls.DockPanel> and <xref:System.Windows.Controls.StackPanel>. This code is run in the `Window.Loaded` event handler:
```

File: dotnet-desktop-guide/framework/wpf/controls/how-to-choose-between-stackpanel-and-dockpanel.md [hunk 2]
```diff
@@ -15,18 +14,24 @@ ms.assetid: f9239086-451f-42e6-81f7-ef89ef349742
 ---
 # How to: Choose Between StackPanel and DockPanel
```
Comment (gewarren) [inline dotnet-desktop-guide/framework/wpf/controls/how-to-choose-between-stackpanel-and-dockpanel.md:15]:
```suggestion
# How to: Choose between StackPanel and DockPanel
```

File: dotnet-desktop-guide/framework/wpf/controls/how-to-choose-between-stackpanel-and-dockpanel.md [hunk 3]
```diff
@@ -15,18 +14,24 @@ ms.assetid: f9239086-451f-42e6-81f7-ef89ef349742
...
+:::image type="content" source="./media/how-to-choose-between-stackpanel-and-dockpanel/example.png" alt-text="A grid with two panels and hearts.":::
 
- The following example demonstrates this key difference between <xref:System.Windows.Controls.DockPanel> and <xref:System.Windows.Controls.StackPanel>.
+## XAML example
 
- [!code-cpp[StackPanelOvw4#1](~/samples/snippets/cpp/VS_Snippets_Wpf/StackPanelOvw4/CPP/StackPanel_Ovw_Sample4.cpp#1)]
- [!code-csharp[StackPanelOvw4#1](~/samples/snippets/csharp/VS_Snippets_Wpf/StackPanelOvw4/CSharp/StackPanel_Ovw_Sample4.cs#1)]
- [!code-vb[StackPanelOvw4#1](~/samples/snippets/visualbasic/VS_Snippets_Wpf/StackPanelOvw4/VisualBasic/StackPanelSamp.vb#1)]
- [!code-xaml[StackPanelOvw4#1](~/samples/snippets/xaml/VS_Snippets_Wpf/StackPanelOvw4/XAML/default.xaml#1)]
+The following example demonstrates this key difference between <xref:System.Windows.Controls.DockPanel> and <xref:System.Windows.Controls.StackPanel> when designing a page in XAML.
```
Comment (gewarren) [inline dotnet-desktop-guide/framework/wpf/controls/how-to-choose-between-stackpanel-and-dockpanel.md:25]:
```suggestion
The following example demonstrates the key difference between <xref:System.Windows.Controls.DockPanel> and <xref:System.Windows.Controls.StackPanel> when designing a page in XAML.
```

Comment (gewarren) [reply dotnet-desktop-guide/framework/wpf/controls/how-to-choose-between-stackpanel-and-dockpanel.md:25]: "this key difference" didn't seem to be appropriate now that it doesn't follow immediately from where that difference is discussed.

File: dotnet-desktop-guide/framework/wpf/controls/how-to-choose-between-stackpanel-and-dockpanel.md [hunk 4]
```diff
@@ -15,18 +14,24 @@ ms.assetid: f9239086-451f-42e6-81f7-ef89ef349742
...
+
+:::code language="xaml" source="./snippets/how-to-choose-between-stackpanel-and-dockpanel/xaml/MainWindow.xaml" id="EmojiViewBox":::
+
+## Code-based example
+
+The following example demonstrates this key difference between <xref:System.Windows.Controls.DockPanel> and <xref:System.Windows.Controls.StackPanel>. This code is run in the `Window.Loaded` event handler:
```
Comment (gewarren) [inline dotnet-desktop-guide/framework/wpf/controls/how-to-choose-between-stackpanel-and-dockpanel.md:31]:
```suggestion
The following example demonstrates the key difference between <xref:System.Windows.Controls.DockPanel> and <xref:System.Windows.Controls.StackPanel>. This code is run in the `Window.Loaded` event handler:
```

**Review by gewarren — APPROVED** [review]

**Pull request merged**